response_writer.go

```diff
@@ -425,3 +426,30 @@ func (w *responseWriter) Push(target string, opts *http.PushOptions) error {
 		return err
 	}
 }
+
+type extWriter struct {
+	rw *responseWriter
+}
+
+func (w *extWriter) Write(p []byte) (n int, err error) {
+	return w.rw.Write(p)
+}
+
+func (w *extWriter) Flush() error {
+	w.rw.Flush()
+	return nil
+}
+
+func (w *extWriter) Finalize() error {
+	w.rw.handlerDone()
+	return nil
+}
+
+func NewResponseWriter(conn network.Conn) network.ExtWriter {
```

Comment:
Better with a return error rather than panic?

Reply:
What is the purpose of this implemented `NewResponseWriter` for?

Reply:
done